akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.javadsl;
+
+import akka.actor.ActorSystem;
+import akka.http.javadsl.HostConnectionPool;
+import akka.japi.Option;
+import akka.japi.Pair;
+import akka.util.ByteString;
+import org.junit.Test;
+
+import scala.Tuple2;
+import scala.concurrent.Future;
+import akka.stream.ActorMaterializer;
+import akka.stream.javadsl.*;
+import akka.http.javadsl.OutgoingConnection;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.Http;
+import scala.util.Try;
+
+public class HttpClientExampleDocTest {
+
+    // compile only test
+    public void testConstructRequest() {
+        //#outgoing-connection-example
+
+        final ActorSystem system = ActorSystem.create();
+        final ActorMaterializer materializer = ActorMaterializer.create(system);
+
+        final Flow<HttpRequest, HttpResponse, Future<OutgoingConnection>> connectionFlow =
+                Http.get(system).outgoingConnection("akka.io", 80);
+        final Future<HttpResponse> responseFuture =
+                Source.single(HttpRequest.create("/"))
+                        .via(connectionFlow)
+                        .runWith(Sink.head(), materializer);
+        //#outgoing-connection-example
+    }
+
+  // compile only test
+  public void testHostLevelExample() {
+    //#host-level-example
+    final ActorSystem system = ActorSystem.create();
+    final ActorMaterializer materializer = ActorMaterializer.create(system);
+
+    // construct a pool client flow with context type `Int`
+    // TODO these Tuple2 will be changed to akka.japi.Pair
+    final Flow<
+      Tuple2<HttpRequest, Integer>,
+      Tuple2<Try<HttpResponse>, Integer>,
+      HostConnectionPool> poolClientFlow =
```

Comment:
sadly it doesn't: 
<img width="933" alt="host-level_client-side_api_ _akka_documentation" src="https://cloud.githubusercontent.com/assets/120979/8766432/4bd8f80e-2e38-11e5-9498-ab3c494b3973.png">